docs/contributing/guidelines.md

```diff
@@ -1,80 +1,117 @@
+Prerequisites: Docker installed and running locally.
+Build the container image:
+
+```bash
+make -f ./builder/Makefile.mkdocs mkdocs-build
+```
+
+Serve the container image:
+
+```bash
+make -f ./builder/Makefile.mkdocs mkdocs-serve
+```
+
+Open `localhost:8000/tracee`
+
+Now you should see the documentation.
+Check that everything is like you intent.
+
+And finally Submit a PR about the changes.
+
+## Contributing to Tracee Code
+
+We welcome contributions to Tracee's codebase! Before submitting your changes, please familiarize yourself with these guidelines.
+To contribute to the code:
+
+### Before You Commit
+
+Tracee relies on several generated files and has strict formatting requirements. Ensure you run the following commands before committing:
+
+1. Man Pages Generation: If you've modified core code or documentation that impacts the man pages, run:
+
+    ```bash
+    make -f builder/Makefile.man
     ```
 
-    ```text
-    Checking C and eBPF files and headers formatting...
-    Checking golang files formatting...
+    This regenerates the man pages to reflect your changes.
+
+2. Protocol Buffer Compilation: If your changes involve modifications to protocol buffer (`.proto`) files,run:
+
+    ```bash
+    make -f builder/Makefile.man
+    ```
+
+    This regenerates the corresponding Go code.
+
+3. Pre-commit checks: Every time you're about to create a pull request, execute:
+
+    ```bash
+    make check-pr
     ```
-    
-    > This will make sure PRs won't fail due to same checks being enforced.
 
-2. Fix Go and C source files formatting
+    This command performs essential checks:
+    - `check-fmt`: Verifies code formatting adheres to project standards.
+    - `check-lint`: Runs linting tools (e.g., `golangci-lint`) to catch potential issues.
+    - `check-code`: Performs static code analysis for both Go and C code.
+    - `format-pr`: Displays the commits in your PR in a standardized format.
 
-!!! fix-fmt Example
+    **Note:** `check-fmt`,`check-lint`,`check-code`,`format-pr` are individual make command combined under `check-pr`
 
-    ```console
+4. Fixing Code Formatting: If `check-fmt` reports issues, use:
+
+    ```bash
     make fix-fmt
     ```
 
-    ```text
-    Fixing C and eBPF files and headers formatting...
-    Formatting ./pkg/ebpf/c/missing_definitions.h
-    Formatting ./pkg/ebpf/c/struct_flavors.h
-    Formatting ./pkg/ebpf/c/tracee.bpf.c
-    Formatting ./pkg/ebpf/c/vmlinux.h
-    
-    Fixing golang files formatting...
-    patching file pkg/ebpf/tracee.go
+    This automatically formats your Go and C code to meet project standards. Review the changes with `git status -s` before committing.
+
+If you don't want to depend on host's libraries versions, or your host is not an **Alpine Linux**/**Ubuntu Linux** or you are using the `alpine-make`/`ubuntu-make` container as a replacement for `make`:
+
+you can run the equivalents checks in a container:
```

Comment:
I believe that's better to guide the contributor to always use VM and/or the docker images for that, so the formatting will be aligned with what we expect. But, a huge but, currently alpine image lack some tooling like clang-format-12 and ubuntu image is crashing when running check-pr - they are focused to building and release process.

We have some specific docker images for helping in the contribute process as:

![image](https://github.com/user-attachments/assets/db27452b-5a6e-4943-90c0-80b5e1829b60)

You might notice that `Makefile.man`, by instance, has a pattern to run the `Dockerfile.man`.

Said that, perhaps bringing a `Dockerfile.checkers` would be the solution here.

Reply:
> Do you say that we should encourage people to run the docker file or container instead of locally?

Yep, as it avoids conflicts with the contributor's host system.

> Also, do you say we should prefer to use the Dockerfiles or that the make file run the docker file?

Makefile should run docker under the hood for us. As it does in protoc, man rules etc.